params/mutations/rewards_test.go

```diff
@@ -192,3 +214,335 @@ func TestGetBlockWinnerRewardForUnclesByEra(t *testing.T) {
+			}
+		}
+		db.Close()
+	}
+}
+*/
+
+func TestGetBlockEra(t *testing.T) {
+	blockNum := big.NewInt(11700000)
+	eraLength := big.NewInt(5000000)
+	era := GetBlockEra(blockNum, eraLength)
+	if era.Cmp(big.NewInt(2)) != 0 {
+		t.Error("Should return Era 2", "era", era)
+	}
+	// handle negative blockNum
+	blockNum = big.NewInt(-50000)
+	era = GetBlockEra(blockNum, eraLength)
+	if era.Cmp(big.NewInt(0)) != 0 {
+		t.Error("Should return Era 0", "era", era)
+	}
+	// handle negative blockNum
+	blockNum = big.NewInt(5000001)
+	era = GetBlockEra(blockNum, eraLength)
+	if era.Cmp(big.NewInt(1)) != 0 {
+		t.Error("Should return Era 1", "era", era)
+	}
+}
+
+func TestGetBlockWinnerRewardByEra2(t *testing.T) {
+	baseReward := big.NewInt(5000000000000000000)
+	era := big.NewInt(0)
+	blockReward := GetBlockWinnerRewardByEra(era, baseReward)
+	if blockReward.Cmp(big.NewInt(5000000000000000000)) != 0 {
+		t.Error("Should return blockReward 5000000000000000000", "reward", blockReward)
+	}
+	era = big.NewInt(1)
+	blockReward = GetBlockWinnerRewardByEra(era, baseReward)
+	if blockReward.Cmp(big.NewInt(4000000000000000000)) != 0 {
+		t.Error("Should return blockReward 4000000000000000000", "reward", blockReward)
+	}
+	era = big.NewInt(2)
+	blockReward = GetBlockWinnerRewardByEra(era, baseReward)
+	if blockReward.Cmp(big.NewInt(3200000000000000000)) != 0 {
+		t.Error("Should return blockReward 3200000000000000000", "reward", blockReward)
+	}
+	era = big.NewInt(3)
+	blockReward = GetBlockWinnerRewardByEra(era, baseReward)
+	if blockReward.Cmp(big.NewInt(2560000000000000000)) != 0 {
+		t.Error("Should return blockReward 2560000000000000000", "reward", blockReward)
+	}
+	era = big.NewInt(4)
+	blockReward = GetBlockWinnerRewardByEra(era, baseReward)
+	if blockReward.Cmp(big.NewInt(2048000000000000000)) != 0 {
+		t.Error("Should return blockReward 2048000000000000000", "reward", blockReward)
+	}
+}
+
+func TestGetRewardForUncle(t *testing.T) {
+	baseReward := big.NewInt(4000000000000000000)
+	era := big.NewInt(0)
+	uncleReward := getEraUncleBlockReward(era, baseReward)
+	if uncleReward.Cmp(big.NewInt(125000000000000000)) != 0 {
+		t.Error("Should return uncleReward 125000000000000000", "reward", uncleReward)
+	}
+	baseReward = big.NewInt(3200000000000000000)
+	uncleReward = getEraUncleBlockReward(era, baseReward)
+	if uncleReward.Cmp(big.NewInt(100000000000000000)) != 0 {
+		t.Error("Should return uncleReward 100000000000000000", "reward", uncleReward)
+	}
+	baseReward = big.NewInt(2560000000000000000)
+	uncleReward = getEraUncleBlockReward(era, baseReward)
+	if uncleReward.Cmp(big.NewInt(80000000000000000)) != 0 {
+		t.Error("Should return uncleReward 80000000000000000", "reward", uncleReward)
+	}
+	baseReward = big.NewInt(2048000000000000000)
+	uncleReward = getEraUncleBlockReward(era, baseReward)
+	if uncleReward.Cmp(big.NewInt(64000000000000000)) != 0 {
+		t.Error("Should return uncleReward 64000000000000000", "reward", uncleReward)
+	}
+}
```

Comment:
duplicate tests to a degree, however I figure more never hurts (ported over from core-pool).